crypto/src/test_utils/zero_crypto.rs

```diff
@@ -0,0 +1,400 @@
+        self.as_ref().partial_cmp(other.as_ref())
+    }
+}
+impl Ord for ZeroPublicKey {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.as_ref().cmp(other.as_ref())
+    }
+}
+
+impl Display for ZeroPublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(f, "{}", Base64::encode_string(self.as_ref()))
+    }
+}
+
+// There is a strong requirement for this specific impl. in Fab benchmarks
+impl Serialize for ZeroPublicKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.encode_base64())
+    }
+}
+
+// There is a strong requirement for this specific impl. in Fab benchmarks
+impl<'de> Deserialize<'de> for ZeroPublicKey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        let s = <String as serde::Deserialize>::deserialize(deserializer)?;
+        let value = Self::decode_base64(&s).map_err(|e| de::Error::custom(e.to_string()))?;
+        Ok(value)
+    }
+}
+
+impl Verifier<ZeroSignature> for ZeroPublicKey {
+    fn verify(&self, _msg: &[u8], _signature: &ZeroSignature) -> Result<(), signature::Error> {
+        Ok(())
+    }
+}
+
+impl<'a> From<&'a ZeroPrivateKey> for ZeroPublicKey {
+    fn from(secret: &'a ZeroPrivateKey) -> Self {
+        let mut hasher = Sha3_256::new();
+        hasher.update(secret.0);
+        let result = hasher.finalize();
+        let bytes: [u8; PUBLIC_KEY_LENGTH] = result[..PUBLIC_KEY_LENGTH]
+            .try_into()
+            .map_err(|_| signature::Error::new())
+            .unwrap();
+        ZeroPublicKey(bytes)
+    }
+}
+
+impl VerifyingKey for ZeroPublicKey {
+    type PrivKey = ZeroPrivateKey;
+    type Sig = ZeroSignature;
+
+    const LENGTH: usize = PUBLIC_KEY_LENGTH;
+
+    fn verify_batch(
+        _msg: &[u8],
+        _pks: &[Self],
+        _sigs: &[Self::Sig],
+    ) -> Result<(), signature::Error> {
+        Ok(())
+    }
+}
+
+///
+/// Implement Authenticator
+///
+
+impl AsRef<[u8]> for ZeroSignature {
+    fn as_ref(&self) -> &[u8] {
+        &[]
+    }
+}
+
+impl std::hash::Hash for ZeroSignature {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.as_ref().hash(state);
+    }
+}
+
+impl PartialEq for ZeroSignature {
+    fn eq(&self, _other: &Self) -> bool {
+        true
+    }
+}
+
+impl Eq for ZeroSignature {}
+
+impl Signature for ZeroSignature {
+    fn from_bytes(_bytes: &[u8]) -> Result<Self, signature::Error> {
+        Ok(ZeroSignature {})
+    }
+}
+
+impl Display for ZeroSignature {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(f, "{}", Base64::encode_string(self.as_ref()))
+    }
+}
+
+impl Authenticator for ZeroSignature {
+    type PubKey = ZeroPublicKey;
+    type PrivKey = ZeroPrivateKey;
+    const LENGTH: usize = 0;
```

Comment:
I'd use at least one byte size vs completely empty sig. I.e. byte zero is a valid sig, and anything else is false or you completely omit checking the byte. I think it would make sense to also have a failed sig to pass some tests in the future.

Reply:
I can add this, but just wondering why we ever care about failed sigs?